src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java

```diff
@@ -0,0 +1,182 @@
+  }
+
+  private ViewModel generateMetadataView(Model model, ApiConfig apiConfig, String template) {
+    int extensionIndex = template.lastIndexOf(".");
+    String outputPath = template.substring(0, extensionIndex);
+
+    return PackageMetadataView.newBuilder()
+        .templateFileName(template)
+        .outputPath(outputPath)
+        .identifier(apiConfig.getDomainLayerLocation())
+        .packageVersion(packageConfig.packageVersion("python"))
+        .protoPath(packageConfig.protoPath())
+        .shortName(packageConfig.shortName())
+        .gaxVersion(packageConfig.gaxVersion("python"))
+        .protoVersion(packageConfig.protoVersion("python"))
+        .commonProtosVersion(packageConfig.commonProtosVersion("python"))
+        .packageName(packageConfig.packageName("python"))
+        .majorVersion(packageConfig.apiVersion())
+        .author(packageConfig.author())
+        .email(packageConfig.email())
+        .homepage(packageConfig.homepage())
+        .license(packageConfig.license())
+        .fullName(model.getServiceConfig().getTitle())
+        .serviceName(apiConfig.getPackageName())
+        .namespacePackages(
+            computeNamespacePackages(apiConfig.getPackageName(), packageConfig.apiVersion()))
+        .build();
+  }
+
+  private List<String> computePackages(String packageName) {
+    ArrayList<String> packages = new ArrayList<>();
+    StringBuilder current = new StringBuilder();
+    boolean first = true;
+    for (String pkg : Splitter.on(".").split(packageName)) {
+      if (!first) {
+        current.append("." + pkg);
+      } else {
+        current.append(pkg);
+        first = false;
+      }
+      packages.add(current.toString());
+    }
+    return packages;
+  }
+
+  private List<String> computeNamespacePackages(String packageName, final String apiVersion) {
+    return Lists.newArrayList(
+        Iterables.filter(
+            computePackages(packageName),
+            new Predicate<String>() {
+              @Override
+              public boolean apply(@Nullable String input) {
+                return isNamespacePackage(input, apiVersion);
+              }
+            }));
+  }
+
+  private boolean isNamespacePackage(String packageName, String apiVersion) {
+    int lastDot = packageName.lastIndexOf(".");
+    return lastDot < 0 || !packageName.substring(lastDot + 1).equals(apiVersion);
+  }
+
+  private List<ViewModel> computeInitFiles(List<String> packages, final String apiVersion) {
+    return Lists.transform(
+        packages,
+        new Function<String, ViewModel>() {
+          @Override
+          @Nullable
+          public ViewModel apply(@Nullable final String packageName) {
+            final String template;
+            if (isNamespacePackage(packageName, apiVersion)) {
+              template = "py/namespace__init__.py.snip";
+            } else {
+              template = "py/__init__.py.snip";
+            }
+            return new ViewModel() {
```

Comment:
Just make a simple ViewModel class with AutoBuilder that can be used here. It feels kind of odd to construct an anonymously-typed object for every instance.